include/clap/ext/state-context.h

```diff
@@ -20,21 +21,27 @@
 ///        clap_plugin_state_context.save(CLAP_STATE_CONTEXT_FOR_PRESET),
 ///        CLAP_STATE_CONTEXT_FOR_PRESET)
 ///
+/// If in doubt, choose CLAP_STATE_CONTEXT_FOR_PRESET as option.
+/// 
 /// If the plugin implements CLAP_EXT_STATE_CONTEXT then it is mandatory to also implement
 /// CLAP_EXT_STATE.
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-static CLAP_CONSTEXPR const char CLAP_EXT_STATE_CONTEXT[] = "clap.state-context.draft/1";
+static CLAP_CONSTEXPR const char CLAP_EXT_STATE_CONTEXT[] = "clap.state-context.draft/2";
 
 enum clap_plugin_state_context_type {
-   // suitable for duplicating a plugin instance
-   CLAP_STATE_CONTEXT_FOR_DUPLICATE = 1,
-
    // suitable for loading a state as a preset
-   CLAP_STATE_CONTEXT_FOR_PRESET = 2,
+   CLAP_STATE_CONTEXT_FOR_PRESET = 1,
+
+  // suitable for duplicating a plugin instance
+   CLAP_STATE_CONTEXT_FOR_DUPLICATE = 2,
+
+   // suitable for loading a state during loading a project/song
+   CLAP_STATE_CONTEXT_FOR_PROJECT = 3,
```

Comment:
Maybe we can add that PROJECT is the identity operation?

Reply:
the documentation said `CLAP_STATE_CONTEXT_FOR_PRESET` is equivalent to `clap_plugin_state->load()`. And it should be kept that way, since `CLAP_STATE_CONTEXT_FOR_PROJECT` has a deeper effect to how the plugin behaves, it does break connections to connected hardware devices and overrides existing ones.  Therefore it should not be default behavior, this should be PREST.